File: DTLiving/Wrapper/VideoFilter.m
//
//  VideoFilter.m
//  DTLiving
//
//  Created by Dan Jiang on 2020/3/17.
//  Copyright © 2020 Dan Thought Studio. All rights reserved.
//

#import "VideoFilter.h"

@interface VideoFilter ()

@end

@implementation VideoFilter

- (instancetype)initWithName:(NSString *)name {
    self = [super init];
    if (self) {
        self.name = name;
    }
    return self;
}

- (NSString *)vertexShaderFile {
    return @"effect_vertex";
}

- (NSString *)fragmentShaderFile {
    return [NSString stringWithFormat:@"%@_fragment", self.name];
}

@end

File: DTLiving/Effect/VideoFilter.swift
//
//  VideoFilter.swift
//  DTLiving
//
//  Created by Dan Jiang on 2020/3/5.
//  Copyright © 2020 Dan Thought Studio. All rights reserved.
//

import CoreMedia

class VideoFilter: VideoOutput, VideoInput {
    
    var nextAvailableTextureIndex: Int {
        return 0
    }
    
    var enabled: Bool {
        return true
    }
    
    func setInputFrameBuffer(_ inputFrameBuffer: FrameBuffer?, at index: Int) {
        
    }
    
    func setInputSize(_ size: CGSize, at index: Int) {
    }
    
    func setInputRotation(_ rotation: VideoRotation, at index: Int) {
    }
    
    func newFrameReady(at time: CMTime, at index: Int) {
    }
    
    func endProcessing() {
    }

}
